### PR TITLE
fix: fix sticky autodetection banner

### LINKED ISSUE #29059: [Bug]: Token detection prompt not dismissing
### Describe the bug

In 12.9.0, the token detection prompt does not dismiss like it did in previous versions.  It used to dismiss regardless of how many tokens you chose to import. Now if you don't import them all, it stays around suggesting the remaining tokens you did not choose to import. 

Not sure if this was intentional.  Compare with 12.8 behavior.

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

Load a new wallet with erc20 tokens.
It should autodetect them.
Only select a subset to import.
The autodetect prompt remains, but maybe should not.

### Error messages or log output

```shell

```

### Detection stage

In production (default)

### Version

12.9.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

fix sticky autodetection banner

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29061?quickstart=1)

## **Related issues**

Fixes: #29059 

## **Manual testing steps**

1. Go to autodetected banner
2. click on ignore all
3. you should have allNetworks selected

### **After**

https://drive.google.com/file/d/11zcmgV1Bdq5wx9MkDlhxr8oDBr9DGB3d/view?usp=sharing

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/detected-token/detected-token.js [hunk 1]
```diff
@@ -210,14 +210,16 @@ const DetectedToken = ({ setShowDetectedTokens }) => {
...
+            allNetworks[chainId];
+          const networkInstanceId =
+            rpcEndpoints[defaultRpcEndpointIndex].networkClientId;
+
+          const tokensToIgnore = tokens.map((token) => token.address);
 
           await dispatch(
             ignoreTokens({
-              tokensToIgnore: tokens,
+              tokensToIgnore,
```
Comment (salimtb) [inline ui/components/app/detected-token/detected-token.js:222]: ignoreTokens should be called with an array of strings

File: ui/components/app/detected-token/detected-token-ignored-popover/detected-token-ignored-popover.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  });
+
+  it('should call handleClearTokensSelection when the confirm button is clicked', () => {
+    renderComponent();
+    const confirmButton = screen.getByTestId(
+      'detected-token-ignored-popover-confirm-button',
+    );
+    fireEvent.click(confirmButton);
+    expect(defaultProps.handleClearTokensSelection).toHaveBeenCalled();
+    expect(defaultProps.handleClearTokensSelection).not.toThrow();
```
Comment (salimtb) [inline ui/components/app/detected-token/detected-token-ignored-popover/detected-token-ignored-popover.test.js:35]: handleClearTokensSelection should not throw an error

File: ui/components/app/detected-token/detected-token-ignored-popover/__snapshots__/detected-token-ignored-popover.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`DetectedTokenIgnoredPopover should match snapshot for ignore mode 1`] = `<DocumentFragment />`;
+
+exports[`DetectedTokenIgnoredPopover should match snapshot for import mode 1`] = `<DocumentFragment />`;
```
Comment (bergeron) [inline ui/components/app/detected-token/detected-token-ignored-popover/__snapshots__/detected-token-ignored-popover.test.tsx.snap:5]: These snapshots are empty, so they aren't testing anything.   I think modals have to be snapshotted differently, I'll see if i can find an example

Comment (bergeron) [reply ui/components/app/detected-token/detected-token-ignored-popover/__snapshots__/detected-token-ignored-popover.test.tsx.snap]: I think this does it:
https://github.com/MetaMask/metamask-extension/pull/29061/commits/0f5eda3569c726401371c338c410e9a358376a7a

**Review by bergeron — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

**Pull request merged**